### PR TITLE
Clean up config-file and option var declarations

### PR DESCRIPTION
**Pull request by cpitclaudel**
Instead of having the `:safe` property on all config file we can just put it once and for all in `def-config-file-var`; and it should allow `nil`.

### CODE REVIEW

File: flycheck.el [hunk 1]
```diff
@@ -6912,7 +6917,7 @@ information about the standard library."
   :type '(choice (const "libc++")
                  (const :tag "GNU libstdc++" "libstdc++")
                  (string :tag "Library name"))
-  :safe #'stringp
+  :safe #'flycheck-string-or-nil-p
```
Comment (fmdkdd) [inline flycheck.el]: The choice list should contain nil.

File: flycheck.el [hunk 2]
```diff
@@ -8374,8 +8375,8 @@ Each item is a string with a tag to be given to `go build'."
 Should be a string representing a version, like 1.6 or 1.11.4.
 See `https://staticcheck.io/docs/#targeting-go-versions' for
 details."
-  :type 'string
-  :safe #'stringp
+  :type '(choice (const :tag "Unspecified" nil)
+                 (string :tag "Version"))
```
Comment (fmdkdd) [inline flycheck.el]: Missing a `-string-or-nil-p`?

Comment (cpitclaudel) [reply flycheck.el]: :eagle: :eye:

File: flycheck.el (hunk 1 above)
Comment (cpitclaudel) [reply flycheck.el]: Good point, fixed.

**Pull request merged**